lib/Service/ParticipantService.php

```diff
@@ -1181,6 +1184,16 @@ public function sendCallNotificationForAttendee(Room $room, Participant $current
 			throw new DoesNotExistException('Room mismatch');
 		}
 
+		$userStatus = $this->userStatusManager->getUserStatuses(array($attendee->getActorId()));
+
+		if (empty($userStatus)) {
+			throw new \InvalidArgumentException('User not found');
+		}
+
+		if ($userStatus[0]->getStatus() === IUserStatus::DND){
+			throw new \InvalidArgumentException('User is on do not disturb mode, failed to send notification');
```

Comment:
I would throw:
```suggestion
			throw new \InvalidArgumentException('status');
```

So you have the reason in the correct format available as `$e->getMessage()` for the last part of https://github.com/nextcloud/spreed/issues/11282#issuecomment-2089819787

Reply:
Thank you for the help, just updated the code with the suggested changes. Also, I'm a little confused with which controller should I add the `return new DataResponse(['error' => 'status'], Http::STATUS_BAD_REQUEST);`. Is it the CallController.php?